src/public/monetization.ts

```diff
@@ -0,0 +1,37 @@
+import { sendMessageToParent } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { SdkError } from './interfaces';
+
+export namespace monetization {
+  /**
+   * @private
+   * Hide from docs
+   * Data structure to represent a subscription plan.
+   */
+  export interface PlanInfo {
+    /**
+     * plan id
+     */
+    planId: string;
+    /**
+     * term of the plan
+     */
+    term: string;
+  }
+
+  /**
+   * @private
+   * Hide from docs
+   * Open dialog to start user's purchase experience
+   * @param callback Callback contains 1 parameters, error.
+   * @param planInfo optional parameter. It contains info of the subscription plan pushed to users.
+   * error can either contain an error of type SdkError, incase of an error, or null when get is successful
+   */
+  export function openPurchaseExperience(callback: (error: SdkError | null) => void, planInfo?: PlanInfo): void {
+    if (!callback) {
+      throw new Error('[open purchase experience] Callback cannot be null');
+    }
+    ensureInitialized();
```

Comment:
Please add a list of frame contexts from which you expect this API to be called (unless it works everywhere). Starting off with just the content frame context is probably a safe option.

Reply:
This API is expected to be called within only Tab or Chat. 

added "FrameContexts.content" inside the initialization. Please check new changes.